frontend/packages/ux-editor/src/components/TextResource/hooks/useGetTextResourceValue.test.ts

```diff
@@ -0,0 +1,39 @@
+import { useGetTextResourceValue } from './useGetTextResourceValue';
+import { textResourceByLanguageAndIdSelector } from '@altinn/ux-editor/selectors/textResourceSelectors';
+import { DEFAULT_LANGUAGE } from 'app-shared/constants';
+import { renderHook } from '@testing-library/react';
+import { type ServicesContextProps } from 'app-shared/contexts/ServicesContext';
+import { ServicesContextProvider } from 'app-shared/contexts/ServicesContext';
+import { type ReactNode } from 'react';
+
+// Test data
+const id = 'testId';
+const value = 'testValue';
+
+// Mocks
+jest.mock('@altinn/ux-editor/hooks', () => ({
+  useTextResourcesSelector: jest.fn((selector) => ({
+    value: selector === textResourceByLanguageAndIdSelector(DEFAULT_LANGUAGE, id) ? value : value,
+  })),
+}));
+
+const renderuseGetTextResourceValue = (props = {}) => {
+  const allProps = {
+    id,
+    ...props,
+  };
+
+  return renderHook(() => useGetTextResourceValue(allProps.id), {
+    wrapper: ({ children }) =>
+      ServicesContextProvider
+        ? ServicesContextProvider({ children } as unknown as ServicesContextProps)
+        : (children as ReactNode),
+  });
+};
```

Comment:
I see that this pattern has been used in previous hook tests 😊 
```suggestion
const renderUseGetTextResourceValue = () => {
  const client = createQueryClientMock();
  return renderHook(() => useGetTextResourceValue(id), {
    wrapper: ({children}) => (
      <ServicesContextProvider {...queriesMock} client={client}>
        {children}
      </ServicesContextProvider>
    ),
  });
};
```